deploy/example.yaml

```diff
@@ -0,0 +1,207 @@
+    nodeSelector:
+    # tolerations can be used if we want the pods of this component
+    # to tolerate some of the taints of nodes.
+    #
+    # Default to nil
+    tolerations:
+
+  # provisioner stores the configuration for OpenEBS provisioner
+  provisioner:
+    enabled: "true"
+    imageTag:
+    replicas: 1
+    nodeSelector:
+    tolerations:
+
+  # localProvisioner stores the configuration for OpenEBS local
+  # provisioner
+  localProvisioner:
+    enabled: "true"
+    imageTag:
+    replicas: 1
+    nodeSelector:
+    tolerations:
+
+  # snapshotOperator stores the configuration for snapshot operator
+  snapshotOperator:
+    enabled: "true"
+    controller:
+      imageTag:
+    provisioner:
+      imageTag:
+    replicas: 1
+    nodeSelector:
+    tolerations:
+
+  # ndm stores the configuration for node-disk-manager
+  ndm:
+    enabled: "true"
+    imageTag: "v0.4.5"
+    replicas: 1
+    sparse:
+      path:
+      size:
+      count:
+    # filters contain the config for excluding or including vendors,
+    # paths, etc.
+    #
+    # Defaults for excluding vendors will be CLOUDBYT,OpenEBS, paths
+    # will be loop,fd0,sr0,/dev/ram,/dev/dm-,/dev/md
+    filters:
+      OSDisk:
+        enabled:
+        exclude: "/,/etc/hosts,/boot"
+      vendor:
+        enabled:
+        exclude: "CLOUDBYT,OpenEBS"
+      path:
+        enabled:
+        exclude: "loop,fd0,sr0,/dev/ram,/dev/dm-,/dev/md"
+        include: ""
+    # probes can be used to configure ndm probes such as Seachest,
+    # smart, capacity, etc.
+    probes:
+      enable:
+      - seachest
+    nodeSelector:
+    tolerations:
+
+  # ndmOperator stores the configuration for ndm operator
+  ndmOperator:
+    enabled: "true"
+    imageTag: "v0.4.4"
+    replicas: 1
+    nodeSelector:
+    tolerations:
+
+  # jiva stores the configuration for CASType: Jiva
+  jiva:
+    imageTag:
+    replicas: 1
+
+  # cstor stores the configuration for CASType: cstor
+  cstor:
+    pool:
+      imageTag:
+    poolMgmt:
+      imageTag:
+    target:
+      imageTag:
+    volumeMgmt:
+      imageTag:
+
+  admissionServer:
+    enabled:
+    imageTag:
+    failurePolicy:
+    replicas:
+    nodeSelector:
+    tolerations:
+    affinity:
+
+  helper:
+    imageTag:
+
+  policies:
+    monitoring:
+      enabled:
+      imageTag:
+
+  analytics:
+    enabled:
+    # Specify in hours the duration after which a ping event needs to be sent.
+    pingInterval: "24h"
+
+  # Options contains the optional flags that can be passed during
+  # installation/upgrade/uninstallation i.e.Timeout can be one of the
+  # optional flags where timeout could be the maximum seconds to wait
+  # at any given step.
+  options:
+status:
+    # Init - Not yet started
+    # InProgress - Installation/Upgrade/UnInstallation is in progress
+    # Error - Installation/Upgrade/UnInstallation resulted in some
+    #         failures
+    # Success - Installation/Upgrade/UnInstallation is successful
+    phase:
```

Comment:
let us have a reason along side phase.
We shall come up with conditions if required.
We can start with Online & Failed phases to being with.